### ISSUE TITLE
Solution ...-ServicesAndWebApps doesn't work

### ISSUE THREAD

**Issue by DenisBabarykin** [dotnet-architecture/eShopOnContainers#394]
I am trying to run *eShopOnContainers-ServicesAndWebApps* solution. 
I have next software on my machine:
* **Windows 10 Pro** 
* **Docker 17.09.0-ce** 
* **Visual Studio 2017 15.3.3** 
* **Dotnet Core 2.0.0**

Solution builds well, but when I start docker from Visual Studio, I get many exceptions in output. Most of them are: *System.Net.Internals.SocketExceptionFactory.ExtendedSocketException*, *RabbitMQ.Client.Exceptions.ConnectFailureException* and *System.Data.SqlClient.SqlException*. Full output log can be seen [here](https://pastebin.com/80A6Tqcg).

Webstatus service shows this:
![webstatus](https://puu.sh/ycpGh/6291fb1002.png)

Do you have any ideas what's wrong? My colleague with nearly the same system configuration has the same problem.

**Comment by CESARDELATORRE** [comment]
@DenisBabarykin  Mmmm, Looks like all the containers with dependencies on SQL have issues in your deployment...
Can you confirm you set Docker up (Docker for Windows setting) with the required memory (at least 4GB) and cores (at least 3) as explained at the Wiki posts?
Looks like that based on errors like the following that I see:

An error occurred while migrating the database used on context PersistedGrantDbContext
System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 35 - An internal exception was caught) ---> System.AggregateException: One or more errors occurred. (No such device or address) ---> System.Net.Internals.SocketExceptionFactory+ExtendedSocketException: No such device or address

Make sure you followed all the steps mentioned at this Wiki post, in the first place:
https://github.com/dotnet-architecture/eShopOnContainers/wiki/02.-Setting-eShopOnContainers-in-a-Visual-Studio-2017-environment

Please, confirm.
Cesar.

**Comment by DenisBabarykin** [comment]
Thanks for reply!
Problem solved! My working machine is Lenovo Thinkpad X260 with:
- Intel Core i5 6200U (4 cores) 
- 8 Gb RAM

I saw in wiki that this solution needed at least 4 Gb RAM for docker, but didn't see info about 3 cores. Therefore I allocated only two cores of CPU and thus got all of these runtime errors and exceptions. After allocation of 3 cores and updating VS to 15.5.0 Preview 3.0 all problems have gone.
Thank you for helping!

**Comment by CESARDELATORRE** [comment]
Welcome! - Still, 8 Gb of RAM for Docker for Windows (Needs Hyper-V) and VS 2017... it is pretty tight as soon as you open additional apps, like Office, etc.
I try to use a Surface Pro with 8 Gb RAM and abandoned because as soon as I was also using other apps, I started getting errors. 
I'd say 12 Gb in minimum if using Docker for Windows and VS 2017 and Office (OUTLOOK, WORD, PowerPoint) at the same time. 
16GB RAM is great. :)

Closing this issue, then.
